src/app/qgsmapsavedialog.cpp

```diff
@@ -270,6 +276,15 @@ void QgsMapSaveDialog::updateOutputSize()
   whileBlocking( mOutputHeightSpinBox )->setValue( mSize.height() );
 }
 
+void QgsMapSaveDialog::checkOutputSize()
+{
+  // check if image size does not exceed QPainter limitation https://doc.qt.io/qt-5/qpainter.html#limitations
+  if ( mSize.width() > 32768 || mSize.height() > 32768 )
+  {
+    mMessageBar->pushWarning( tr( "Save as image" ), tr( "Output raster will be truncated, as image width or height is larger than 32768 pixels." ) );
```

Comment:
```suggestion
    mMessageBar->pushWarning( QString(), tr( "Output raster will be truncated, as image width or height is larger than 32768 pixels." ) );
```

That'll save a few more unnecessary characters!